binder/src/test/java/io/grpc/binder/BinderTransportSecurityTest.java

```diff
@@ -0,0 +1,235 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.binder;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.common.util.concurrent.Uninterruptibles;
+import com.google.protobuf.Empty;
+
+import android.app.Application;
+import android.content.ComponentName;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.IBinder;
+import android.os.Looper;
+
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.protobuf.lite.ProtoLiteUtils;
+
+import androidx.lifecycle.LifecycleService;
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.android.controller.ServiceController;
+
+import java.io.IOException;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.Executor;
+
+import io.grpc.MethodDescriptor;
+import io.grpc.Server;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerMethodDefinition;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.stub.ClientCalls;
+import io.grpc.stub.ServerCalls;
+
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+public final class BinderTransportSecurityTest {
+
+  private static final String SERVICE_NAME = "fake_service";
+  private static final String FULL_METHOD_NAME = "fake_service/fake_method";
+  private final Application context = ApplicationProvider.getApplicationContext();
+  private ServiceController<SomeService> controller;
+  private SomeService service;
+  private ManagedChannel channel;
+
+  @Before
+  public void setUp() {
+    controller = Robolectric.buildService(SomeService.class);
+    service = controller.create().get();
+
+    AndroidComponentAddress listenAddress = AndroidComponentAddress.forContext(service);
+    channel = BinderChannelBuilder.forAddress(listenAddress, context).build();
```

Comment:
I worry this is going to be flaky unless you control the Channel's executor(), scheduledExecutor() and offloadExecutor(), installed here.

Reply:
Done.